lib/solveEquation/stepThrough.js

```diff
@@ -68,7 +71,42 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
     }
 
     try {
```

Comment:
this is all in a try block - I think it'll catch a lot more errors than you're thinking for a lot of cases you're not considering. Maybe you want to do that for now, but it seems pretty not great to me - any error could happen, even bugs in your code, and you wouldn't know